Sources/TuistEnvKit/Settings/VersionProvider.swift

```diff
@@ -16,98 +16,63 @@ protocol VersionProviding {
 }
 
 enum VersionProviderError: FatalError {
-    case dataDecodingError(url: Foundation.URL)
-    case responseError(url: Foundation.URL, content: String, statusCode: Int)
+    case noVersionsError
 
     var description: String {
         switch self {
-        case let .dataDecodingError(url):
-            return "Error decoding the response from \(url.absoluteString) as an utf8 string."
-        case let .responseError(url, content, statusCode):
-            return """
-            The request to \(url.absoluteString) return an unsuccessful response with status code \(statusCode) and error body:
-            \(content)
-            """
+        case .noVersionsError:
+            return "Error fetching versions from git."
         }
     }
 
     var type: ErrorType {
         switch self {
-        case .dataDecodingError: return .bug
-        case .responseError: return .bug
+        case .noVersionsError: return .bug
         }
     }
 }
 
 class VersionProvider: VersionProviding {
-    let requestDispatcher: HTTPRequestDispatching
+    let gitHandler: GitHandling
 
-    init(requestDispatcher: HTTPRequestDispatching = HTTPRequestDispatcher()) {
-        self.requestDispatcher = requestDispatcher
+    init(gitHandler: GitHandling = GitHandler()) {
+        self.gitHandler = gitHandler
     }
 
     func versions() -> AnyPublisher<[Version], Error> {
-        requestDispatcher.dispatch(resource: changelogResource())
-            .flatMapLatest { content, _ -> AnyPublisher<[Version], Error> in
-                do {
-                    let versions = try self.parseVersionsFromChangelog(content)
-                    return AnyPublisher(value: versions)
-                } catch {
-                    return AnyPublisher(error: error)
-                }
-            }
-            .eraseToAnyPublisher()
+        do {
+            let content = try gitHandler.lsRemote(url: Constants.gitRepositoryURL)
+            let versions = try parseVersionsFromGit(content)
+            return AnyPublisher(value: versions)
```

Comment:
I don't think there is a point of using `AnyPublisher` here? It should either be converted to actually do the work asynchronously or we can just keep synchronous and remove `AnyPublisher` completely here.